### DISCUSSION TITLE
Constant value column / Single row table

### DISCUSSION THREAD

**Discussion by gruvw** [simolus3/drift#3315] [Q&A] [answered]
Is it possible to declare a constant column in a table ? (like constant int of value 0 let's say)

That way, I would be able to set that column as the primary key (unique) of the table and enforce a single row table.

Something like:

```dart
IntColumn get unique => integer().constant(0)(); // ?

@override
Set<Column> get primaryKey => {unique};
```

**Comment by simolus3** [comment, marked as answer]
That's not possible, but you can set a default and a check which effectively does the same thing:

```dart
IntColumn get unique => integer().withDefault(const Constant(0)).check(unique.equals(0))();

@override
Set<Column> get primaryKey => {unique};
```
